flask_compress/flask_compress.py

```diff
@@ -195,14 +201,14 @@ def after_request(self, response):
 
         response.set_data(compressed_content)
 
-        response.headers['Content-Encoding'] = chosen_algorithm
-        response.headers['Content-Length'] = response.content_length
+        response.headers["Content-Encoding"] = chosen_algorithm
+        response.headers["Content-Length"] = response.content_length
 
         # "123456789"   => "123456789:gzip"   - A strong ETag validator
         # W/"123456789" => W/"123456789:gzip" - A weak ETag validator
-        etag = response.headers.get('ETag')
+        etag = response.headers.get("ETag")
         if etag:
-            response.headers['ETag'] = '{0}:{1}"'.format(etag[:-1], chosen_algorithm)
+            response.headers["ETag"] = f'{etag[:-1]}:{chosen_algorithm}"'
```

Comment:
This caught my eye; the before and after versions of this code result in an ETag string like

```
xyz"
```

I'm pretty sure there should be a double quote at the beginning, too.

Reply:
There should be a test for this, but the code is actually fine here.
It was committed 4 years ago in https://github.com/colour-science/flask-compress/commit/9dadd688d2f1d1000e2530e0f71017ccc681a884

There is a comment just above the code, there are 2 types of Etags:

```
# "123456789"   => "123456789:gzip"   - A strong ETag validator
# W/"123456789" => W/"123456789:gzip" - A weak ETag validator
```
The current code works fine with both:

```python
>>> etag = '"123456789"'
>>> '{0}:{1}"'.format(etag[:-1], "gzip")
'"123456789:gzip"'
>>>
>>> etag = 'W/"123456789"'
>>> '{0}:{1}"'.format(etag[:-1], "gzip")
'W/"123456789:gzip"'
```
I agree this is not super elegant, but it works, and I do not think we can make it much faster.

Reply:
OH! Yep, it's removing the trailing quote and then adding it back in. Makes sense. :+1: